recipes/highs/all/conanfile.py

```diff
@@ -0,0 +1,86 @@
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import apply_conandata_patches, collect_libs, export_conandata_patches, get
+from conan.tools.microsoft import is_msvc
+
+required_conan_version = ">=1.53.0"
+
+
+class HiGHSConan(ConanFile):
+    name = "highs"
+    description = "high performance serial and parallel solver for large scale sparse linear optimization problems"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.highs.dev/"
+    topics = ("highs", "simplex", "interior point", "solver", "linear", "programming")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def validate(self):
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
```

Comment:
Usually with cmake it's just a matter of enabling export of symbols (I did not look at the history but just maybe if you didn't try it couldn't hurt to try)

Reply:
I tried building it as shared lib on MSVC. Here in the CI I get a [failed pipeline without an error message](https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/15814/8-configs/windows-visual_studio/highs/1.4.2//summary.json) that helps me. On my local machine I get a LNK2019 error "unresolved external symbol". Even enabling `WINDOWS_EXPORT_ALL_SYMBOLS` doesn't fix that.